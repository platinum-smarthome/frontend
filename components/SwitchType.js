import React, { Component } from 'react';
import { View, Image, Text, StyleSheet, Switch } from 'react-native'
import CO2Status from './status/CO2Status'
import GarageStatus from './status/GarageStatus'
import DoorStatus from './status/DoorStatus'
import { updateSensorStatus } from '../store/sensors/sensor.actions'
import { connect } from 'react-redux'

class SwitchType extends Component {
  render() {
    return (
      <View style={styles.switchTypes}>
        <View>
          <Image style={styles.switchImg} source={ this.props.imgLogo }/>
        </View>
        <View style={styles.textContent}>
          <Text style={styles.switchText}> { this.props.text } </Text>
          { this.props.sensors.loading ?
            <Text> </Text> :
            this.props.status ? 
            <Text style={styles.locked}> On</Text> :
            <Text style={styles.unlocked}> Off</Text>
          }
        </View>
        <View style={styles.switchPos}>
        { this.props.sensors.loading ?
          ( <Image style={{width: 60, height: 60, marginTop: -16}} source={require('../components/assets/spinner.gif')} /> ) :
          // ( <Text> Loading </Text> ) : 
          ( <Switch onValueChange={(e) => this.props.updateSensorStatus({type: this.props.type, value: e })} value={ (this.props.status) ? true : false }/> )
        }
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  switchTypes: {
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: '#d3d3d3',
  },
  switchImg: {
    width: 64,
    height: 64,
    marginHorizontal: 15,
    marginVertical: 6
  },
  textContent: {
    paddingVertical: 10,
    flexDirection: 'column',
  },
  switchText: {
    fontSize: 22,
    fontWeight: '300',
    color: '#333333'
  },
  switchPos: {
    position: 'absolute',
    right: 0,
    paddingHorizontal: 20,
    paddingVertical: 20
  },
  locked: {
    fontSize: 16,
    fontWeight: '300',
    color: '#006ac6'
  },
  unlocked: {
    fontSize: 16,
    fontWeight: '300',
    color: 'red'
  }
})

function mapStateToProps (state) {
  return {
    sensors: state.sensors
  }
}

function mapDispatchToProps (dispatch) {
  return {
    updateSensorStatus: (payload) => dispatch(updateSensorStatus(payload))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SwitchType)