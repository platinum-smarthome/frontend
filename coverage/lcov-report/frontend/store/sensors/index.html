<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for frontend/store/sensors</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> frontend/store/sensors
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>28/49</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.84% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.36% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>27/44</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="sensor.actionType.js"><a href="sensor.actionType.js.html">sensor.actionType.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file medium" data-value="sensor.actions.js"><a href="sensor.actions.js.html">sensor.actions.js</a></td>
	<td data-value="51.43" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 51%;"></div><div class="cover-empty" style="width:49%;"></div></div></td>
	<td data-value="51.43" class="pct medium">51.43%</td>
	<td data-value="35" class="abs medium">18/35</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="18" class="abs low">6/18</td>
	<td data-value="56.67" class="pct medium">56.67%</td>
	<td data-value="30" class="abs medium">17/30</td>
	</tr>

<tr>
	<td class="file medium" data-value="sensor.reducers.js"><a href="sensor.reducers.js.html">sensor.reducers.js</a></td>
	<td data-value="55.56" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 55%;"></div><div class="cover-empty" style="width:45%;"></div></div></td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Apr 25 2018 17:39:42 GMT+0700 (WIB)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
