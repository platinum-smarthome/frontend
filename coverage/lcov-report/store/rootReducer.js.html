<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for store/rootReducer.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">store</a> rootReducer.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { combineReducers } from 'redux'
&nbsp;
import Data from './data/data.reducers'
import CreateNewHome from './createNewHome/createNewHome.reducers'
import AddNewUser from './addNewUser/addNewUser.reducers'
import Sensor from './sensors/sensor.reducers'
import HomeData from './homeData/homeData.reducers'
import UserData from './userData/userData.reducers'
import NotificationLogs from './notificationLogs/notificationLogs.reducers'
import HousePin from './housePin/housePin.reducers'
import MemberList from './memberList/memberList.reducers'
import Alarms from './alarms/alarms.reducers'
&nbsp;
const reducers = combineReducers({
  addNewUserReducer: AddNewUser,
  data: Data,
  CreateNewHomeReducer: CreateNewHome,
  sensors: Sensor,
  HomeData: HomeData,
  UserData: UserData,
  NotificationLogs: NotificationLogs,
  housePin: HousePin,
  MemberList: MemberList,
  AlarmsData : Alarms
})
&nbsp;
export default reducers</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Apr 25 2018 13:02:35 GMT+0700 (WIB)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
<script src="../block-navigation.js"></script>
</body>
</html>
