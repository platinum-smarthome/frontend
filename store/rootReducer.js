import { combineReducers } from 'redux'
import Data from './data/data.reducers'
import CreateNewHome from './createNewHome/createNewHome.reducers'
import AddNewUser from './addNewUser/addNewUser.reducers'
import HomeData from './homeData/homeData.reducers'
import UserData from './userData/userData.reducers'

const reducers = combineReducers({
  addNewUserReducer: AddNewUser,
  data: Data,
  CreateNewHomeReducer: CreateNewHome,
  HomeData: HomeData,
  UserData: UserData
})

export default reducers