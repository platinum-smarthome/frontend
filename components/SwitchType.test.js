import React, { Component } from 'react'
import enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import store from '../store/store'

import SwitchType from './SwitchType'
import { Text, View } from 'react-native'

enzyme.configure({ adapter: new Adapter()})

let wrapper
let mockFunction

beforeEach(() => {
  wrapper = shallow(<SwitchType store={store}/>)
  mockFunction = jest.fn()
})


describe('<SwitchType/>', () => {
  it('should render <SwitchType/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('Loading Test', () => {
  it('should render empty <Text>', () => {
    let wrap = shallow(<SwitchType store={store}/>)
    expect(wrap.find(<Text> </Text>)).toBeDefined()
  })
})
// describe('<SwitchType/> child is rendered', () => {
//   it('Should render <View/>', () => {
//     expect(wrapper.find(<View/>)).toBeDefined()
//     expect(wrapper.find('View').length).toBe(5)  
//   })
//   it('Should render <Text/>', () => {
//     expect(wrapper.find(<Text/>)).toBeDefined()
//     expect(wrapper.find('Text').length).toBe(5)  
//   })
// })

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<SwitchType store={store}/>)
    expect(tree).toMatchSnapshot()
  })
})