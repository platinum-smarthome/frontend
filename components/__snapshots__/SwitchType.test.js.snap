// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#d3d3d3",
      "borderBottomWidth": 1,
      "flexDirection": "row",
    }
  }
>
  <View>
    <Image
      source={undefined}
      style={
        Object {
          "height": 64,
          "marginHorizontal": 15,
          "marginVertical": 6,
          "width": 64,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingVertical": 4,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
          "fontSize": 22,
          "fontWeight": "300",
        }
      }
    >
       
       
    </Text>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "red",
            "fontSize": 16,
            "fontWeight": "300",
          }
        }
      >
         
        Off
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#d3d3d3",
            "fontSize": 14,
            "fontWeight": "300",
          }
        }
      >
         Unlock Main Door to enable switch 
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 20,
        "paddingVertical": 20,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <RCTSwitch
      disabled={true}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          undefined,
        ]
      }
      value={false}
    />
  </View>
</View>
`;
